File: SP-API/Domain/ItemBatch.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class ItemBatch
    {
        public int Id { get; set; }
        public int ItemId { get; set; }
        public int ItemStatusId { get; set; }
        public int StockCount { get; set; }
        public int ItemCount { get; set; }
        public int ReorderCount { get; set; }
        public string ItemBatchCode { get; set; }
        public string Name { get; set; }
        public decimal MaxRetailPrice { get; set; }
        public DateTime ManufactureDate { get; set; }
        public DateTime ExpiryDate { get; set; }
        public virtual Item Item { get; set; }
        public virtual Status Status { get; set; }
        public virtual ICollection<OrderItemBatch> OrderItemBatches { get; set; }
    }
}

File: SP-Client/src/app/_models/itemBatch.ts
export interface IItemBatchEnvelope {
  itemBatches: ItemBatch[];
  recordCount: number;
}

export class ItemBatch {
  id: number;
  itemId: number;
  itemName: string;
  itemStatusId: number;
  itemStatusName: string;
  itemCount: number;
  stockCount: number;
  reorderCount: number;
  name: string;
  itemBatchCode: string;
  maxRetailPrice: number;
  manufactureDate: string;
  expiryDate: string;

  constructor(itemBatch) {
    this.id = itemBatch.id;
    this.itemId = itemBatch.itemId;
    this.itemName = itemBatch.itemName;
    this.itemStatusId = itemBatch.itemStatusId;
    this.itemStatusName = itemBatch.itemStatusName;
    this.itemCount = itemBatch.itemCount;
    this.stockCount = itemBatch.stockCount;
    this.reorderCount = itemBatch.reorderCount;
    this.name = itemBatch.name;
    this.itemBatchCode = itemBatch.itemBatchCode;
    this.maxRetailPrice = itemBatch.maxRetailPrice;
    this.manufactureDate = itemBatch.manufactureDate;
    this.expiryDate = itemBatch.expiryDate;
  }
}
